src/parser.rs

```diff
@@ -188,13 +227,12 @@ impl<'i, 't> Parser<'i, 't> {
     ///
     /// This ignores whitespace and comments.
     #[inline]
-    pub fn expect_exhausted(&mut self) -> Result<(), ()> {
+    pub fn expect_exhausted(&mut self) -> Result<(), BasicParseError<'i>> {
         let start_position = self.position();
         let result = match self.next() {
-            Err(()) => Ok(()),
-            Ok(_) => {
-                Err(())
-            }
+            //FIXME: swallowing non-UnexpectedEof errors seems wrong, but tests fail otherwise.
```

Comment:
Huh... Could you file an issue with more details, and perhaps an example, and reference it here, just to see if it's something expected or not? Which other errors are reported?

(also, uber-nit: a space after `//`)